docs/guides/shortnames.md

```diff
@@ -0,0 +1,164 @@
+When a Floyd CLI command takes a job name, a dataset name, or a job output, you can always specify a full length name, like this:
+
+Job:
+```
+$ floyd status mckay/projects/quick-start/1
+```
+
+Dataset:
+```
+$ floyd data status mckay/datasets/mnist/1
+```
+
+Job Output:
+```
+$ floyd status mckay/projects/quick-start/1/output
+```
+
+Floyd CLI allows you to use shortened names instead of full-length ones. When
+you leave out parts of a name, the CLI does the following:
+
+1. When a username is missing, the username of the logged-in user is used.
+2. When a project or dataset name is missing, the name of the currently
+    initialized project or dataset is used.
+3. When a job number or a dataset version number is missing, the most recent version is used.
+4. If you leave out the `/projects/` or `/datasets/` part of the name, it is
+    inferred based on the command and other parts of the name. In most cases,
+    you shouldn't have to specify these parts of the name. However, if you want
+    to specify an output, you'll need to make sure your name has `/output` at the
+    end of it, output names are otherwise the same as job names.
+
+These assumptions allow you to avoid typing out a full name in almost any
+situation.
+
+## Examples
+
+Here are some examples that demonstrate how to use shortnames. For these
+examples, we'll assume your username is `fooster`, the project you've
+initialized in the current directory is called `my_project`, and the last job
+you ran in `my_project` is number `3`. Each example will show two code
+snippets: the first showing a command using a shortened name, and a second
+showing the full-length equivalent of the first command.
+
+
+## Job Name Examples
```

Comment:
This is great. Can you also add the list of subcommands where this shortened name would work? preferably with links to their usage pages?